packages/pwa-kit-extension-sdk/src/react/hooks/useApplicationExtensionsStore.tsx

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2025, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {create} from 'zustand'
+import {devtools} from 'zustand/middleware'
+
+// export type SliceInitializer<T> = (set: (partial: any) => void, get: () => BaseStore) => T
+export type SliceInitializer<T> = (
+    set: (partial: (state: T) => Partial<T>) => void,
+    get: () => T
+) => T
+
+interface BaseStore {
+    state: Record<string, any>
+    addSlice: <T>(name: string, initializer: SliceInitializer<T>) => void
+    getSlice: <T>(name: string) => T | undefined
+}
+
+/**
+ * Zustand store for managing application extensions with dynamic slices.
+ * Provides methods to add and retrieve slices of state dynamically.
+ */
+export const useApplicationExtensionsStore = create<BaseStore>()(
+    devtools((set, get) => ({
```

Comment:
nice, built-in devtools support.

Do we want to add a flag for it in case people hate it on production?

Reply:
Looking at the implementation of the devtools middleware it already does that for us.

Reply:
ah nice, shall we add a comment? I feel like I won't be the only guy who ask this question